bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,128 @@
+import scala.concurrent.Future
+import scalaz._
+import Scalaz._
+
+/**
+ * Represents an AGS strategy consisting of multiple substrategies acting together.
+ */
+sealed case class MultiProbeStrategy(key: AgsStrategyKey, strategies: List[AgsStrategy]) extends AgsStrategy {
+
+  override def magnitudes(ctx: ObsContext, mt: MagnitudeTable): List[(GuideProbe, AgsMagnitude.MagnitudeCalc)] =
+    strategies.flatMap(_.magnitudes(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable): List[AgsAnalysis] =
+    strategies.flatMap(_.analyze(ctx, mt))
+
+  override def analyze(ctx: ObsContext, mt: MagnitudeTable, guideProbe: ValidatableGuideProbe, guideStar: SiderealTarget): Option[AgsAnalysis] =
+    AgsAnalysis.analysis(ctx, mt, guideProbe, guideStar, probeBands)
+
+  override def catalogQueries(ctx: ObsContext, mt: MagnitudeTable): List[CatalogQuery] =
+    strategies.flatMap(_.catalogQueries(ctx, mt))
+
+  override def candidates(ctx: ObsContext, mt: MagnitudeTable): Future[List[(GuideProbe, List[SiderealTarget])]] =
+    Future.traverse(strategies)(_.candidates(ctx, mt)).map(_.flatten)
+
+  override def estimate(ctx: ObsContext, mt: MagnitudeTable): Future[AgsStrategy.Estimate] =
+    Future.fold(strategies.map(_.estimate(ctx, mt)))(1.0)((p,est) => p * est.probability).map(Estimate(_))
+
+  // This is a bit hacked, as it requires invoking select for each of the substrategies, which are structured to
+  // pick the best assignment for the current position angle constraint. This, of course, might result in different
+  // position angles for substrategies if we naively call select on each, so instead, we hack the context to
+  // handle each position angle constraint and then pick the best one.
+  override def select(ctx: ObsContext, mt: MagnitudeTable): Future[Option[AgsStrategy.Selection]] = {
+    // We hack the obs context to be a list of obs contexts with all possible position angles, and with fixed
+    // position angle constraint so that guide probes will not be allowed to take on different angles in select calls.
+    def ctxToFixedList: List[ObsContext] = {
+      val ctxFixed = {
+        val instFixed = ctx.getInstrument match {
+          case pac : SPInstObsComp with PosAngleConstraintAware =>
+            val newPac = pac.clone.asInstanceOf[SPInstObsComp with PosAngleConstraintAware]
+            newPac.setPosAngleConstraint(PosAngleConstraint.FIXED)
+            newPac
+          case oth : SPInstObsComp => oth
+        }
+        ctx.withInstrument(instFixed)
+      }
+      (ctx.getPosAngleConstraint match {
+        case PosAngleConstraint.FIXED                                            => List(0)
+        case PosAngleConstraint.FIXED_180 | PosAngleConstraint.PARALLACTIC_ANGLE => List(0, 180)
+        case PosAngleConstraint.UNBOUNDED                                        => Range(0, 360, 15).toList
+      }).map(a => ctxFixed.withPositionAngle(ctx.getPositionAngle.add(a, Angle.Unit.DEGREES)))
```

Comment:
How did you come up with 15º increments for the `UNBOUNDED` case?  I notice that `SingleProbeStrategy` has a default and minimum of 10º (see `SingleProbeStrategyParams.stepSize`) but it can vary.  I don't think we have any way in general to go from `AgsStrategy` to an appropriate step size (in order to, say, pick the minimum of all the strategies) so maybe for now we should make the increment 10º and add a task to generalize this correctly in the future.